pavelib/stanford_i18n.py

```diff
@@ -0,0 +1,282 @@
+"""
+Stanford internationalization tasks
+"""
+
+from __future__ import absolute_import
+
+import os
+import shutil
+import subprocess as sp
+
+from i18n.config import Configuration
+from i18n import segment
+from path import Path
+from paver.easy import task, cmdopts, needs, sh
+import polib
+from .utils.cmd import django_cmd
+from .utils.stanford_i18n_helpers import fix_header, fix_metadata, segment_pofile_lazy
+
+BASE_DIR = Path('.').abspath()
+CONFIG = Configuration(filename=BASE_DIR / 'conf/locale/stanford_config.yaml')
+
+@task
+@needs(
+    "pavelib.prereqs.install_prereqs",
+)
+def stanfordi18n_extract_platform():
+
+    files_to_clean = set()
+
+    babel_cmd_template = (
+        'pybabel extract --mapping={config} '
+        '--add-comments="Translators:" --keyword="interpolate" '
+        '. --output={output}'
+    )
+    babel_mako_cfg = CONFIG.locale_dir.joinpath('stanford_mako.cfg')
+    mako_file = CONFIG.source_messages_dir.joinpath('stanford_mako.po')
+    babel_mako_cmd = babel_cmd_template.format(
+        config=babel_mako_cfg,
+        output=mako_file,
+    )
+    print babel_mako_cmd
+    sp.check_call(babel_mako_cmd, shell=True)
+    files_to_clean.add(mako_file)
+
+    babel_underscore_cfg = CONFIG.locale_dir.joinpath('stanford_underscore.cfg')
+    underscore_file = CONFIG.source_messages_dir.joinpath('stanford_underscore.po')
+    babel_underscore_cmd = babel_cmd_template.format(
+        config=babel_underscore_cfg,
+        output=underscore_file,
+    )
+    print babel_underscore_cmd
+    sp.check_call(babel_underscore_cmd, shell=True)
+    files_to_clean.add(underscore_file)
+
+    # Rename django.po and djangojs.po to avoid being overwritten
+    os.rename(CONFIG.source_messages_dir.joinpath('django.po'), CONFIG.source_messages_dir.joinpath('django-saved.po'))
+    os.rename(CONFIG.source_messages_dir.joinpath('djangojs.po'), CONFIG.source_messages_dir.joinpath('djangojs-saved.po'))
+
+    makemessages = "django-admin.py makemessages -l en"
+    ignores = " ".join('--ignore="{}/*"'.format(d) for d in CONFIG.ignore_dirs)
+    if ignores:
+            makemessages += " " + ignores
+    print makemessages
+    sp.check_call(makemessages, shell=True)
+
+    django_file = CONFIG.source_messages_dir.joinpath('stanford_django.po')
+    os.rename(CONFIG.source_messages_dir / 'django.po', django_file)
```

Comment:
Here we're doing `CONFIG.source_messages_dir / 'django.po'` and above we have `CONFIG.source_messages_dir.joinpath('django.po')` are these meant to be equivalent?
If so, we should probably use the same syntax, or even better, memoize to a local variable and use that value throughout, instead of continually recalculating the path.

Reply:
Or maybe even better still, create a helper:

```python
def rename(name_source, name_destination):
    path_source = CONFIG.source_messages_dir / name_source
    path_destination = CONFIG.source_messages_dir / name_destination
    os.rename(path_source, path_destination)
    return path_source, path_destination
```

Reply:
There are a couple of options, but I think we should definitely standardize how we're doing this everywhere and try to reduce as much duplication as possible.
It's harder to follow, as is, and is more prone to mistakes.